dashboard/app/models/user.rb

```diff
@@ -126,25 +126,42 @@ class User < ActiveRecord::Base
   # You can be associated with districts through cohorts
   # has_many :districts, through: :cohorts.
 
+  # @param permission [UserPermission] the permission to query.
+  # @return [Boolean] whether the User has permission granted.
+  # TODO(asher): Determine whether request level caching is sufficient, or
+  #   whether a memcache or otherwise should be employed.
+  def permission_from_cache?(permission)
+    # Default to using the value cached in the instance variable, if present.
+    unless @permissions.nil?
+      return @permissions.include? permission
+    end
+    # The user's permissions were not cached, so query DB and save to instance
+    # variable.
+    @permissions = UserPermission.where(user_id: id).pluck(:permission)
+    return @permissions.include? permission
+  end
+
   def facilitator?
-    permission? UserPermission::FACILITATOR
+    permission_from_cache?(UserPermission::FACILITATOR)
```

Comment:
There are lots of callers of `permission?` outside of this file, all of which will get cached behavior because you changed `permission?` to read from cache. For consistency, maybe **don't** change `facilitator` and `workshop_organizer` and have them continue to call `permission?`? Seems like we should do it consistently, and change all or change none.